applications/plugins/ArticulatedSystemPlugin/src/ArticulatedSystemPlugin/ArticulatedHierarchyContainer.inl

```diff
@@ -48,6 +49,7 @@ ArticulationCenter::ArticulationCenter():
     posOnChild(initData(&posOnChild, "posOnChild", "Child position of the articulation center")),
     articulationProcess(initData(&articulationProcess, (int) 0, "articulationProcess", " 0 - (default) hierarchy between articulations (euler angles)\n 1- ( on Parent) no hierarchy - axis are attached to the parent\n 2- (attached on Child) no hierarchy - axis are attached to the child"))
 {
+    d_componentState.setValue(sofa::core::objectmodel::ComponentState::Loading);
```

Comment:
Is it set to `valid` at some point?

Reply:
Thank for your question.

Up to know it is considered that init() is the only place to flip an object to the Valid state. So this correspond to the line Line 216 in ArticulatedHierarchyContainer.cpp (in the init() method).

Reply:
But the line you are referring to (in the inl file right?) initializes the class `ArticulatedHierarchyContainer`, not `ArticulationCenter`.
In a future PR, all the classes must be in dedicated files.